### ISSUE TITLE
ScreenScraper API is down/unstable at the moment

### ISSUE THREAD

**Issue by waugh-hoo** [muldjord/skyscraper#187]
I'm receiving a new error with screenscraper.  It states that the json file is empty.  I was able to change to the GamesDB and it did successfully scrape, so I suspect the software is working correctly.  I'm not very savvy in this department so forgive me for being brief.  I am able to provide anything you need, I'm just not sure what details are required for you to troubleshoot.

**Comment by waugh-hoo** [comment]
went and got the screenscraper_error.json file in hopes that it would provide some detail, however it was empty.

**Comment by waugh-hoo** [comment]
In reading further in the similar bugs that were posted before, I see that this possibly is an indication that the API is down.  I'll keep trying here and if I still don't have success, I'll update or close if it works.

**Comment by waugh-hoo** [comment]
I did some more digging and found an issue regarding ScreenScraper requiring an account.  I created an account and adjusted the config.ini file so that it has the credentials I set up for ScreenScraper and tried again.  What I got was that SkyScraper:

- Fetching limits for 'user'
- received invalid ScreenScraper server response, maybe their server is having issues, forcing one thread...

and the conclusion to this paragraph was...
- successfully parsed 0 resources!

Can I assume this means that the screenscraper server is the issue and that it is working with the credentials I entered and I will just have to try again at some point?

**Comment by muldjord** [comment]
ScreenScraper has been down for a while now, that's the reason. We'll just have to wait until they get their API back up and running again. It's been very unstable over the last few days it seems. It works right now, but it's very slow. I suspect they are working to fix it.

Closing.

**Comment by muldjord** [comment]
I've reopened the issue for others to see and updated the title as it was slightly misleading to the actual cause.

**Comment by waugh-hoo** [comment]
Thanks muldjord.  I appreciate the confirmation.  I had just used skyscraper about a week prior for a Pi3B+ retropie build and was surprised I was having an issue on my MintyPi build (piZero).  It's good to know that there was an external factor causing an issue.

**Comment by waugh-hoo** [comment]
is there a way that I can move over the scraped data from my existing build and have Skyscraper use it on the new build?  I've found the data on the existing build, but just moving it over does not allow it to show up even after having skyscraper rebuild the game lists on the new machine.

**Comment by muldjord** [comment]
Yes, you can easily do that. You can read all about the cache here: https://github.com/muldjord/skyscraper/blob/master/docs/CACHE.md

Install Skyscraper on the other machine and then you just need to move the `~/.skyscraper/cache` folder to the exact same location on the other machine. As long as the roms are the same you can generate the gamelist with `Skyscraper -p PLATFORM` on the other machine. If it doesn't work I'm guessing you haven't placed it in the correct location.

**Comment by waugh-hoo** [comment]
I found the cache documents and I'm playing around in there now.  I was using the source as local, and had limited success, but yet to have it move the videos/box art/screen capture yet.  The XML data did come across well though.  I'm still messing about.  I'll keep at it.  Thanks for the input.

**Comment by muldjord** [comment]
Just be very careful not to change the files or folder structures inside the cache folder. Otherwise you will break the link between the data and the media files. Best to just copy the entire cache folder over as it is without changing anything inside of it.

EDIT: If you've moved some of the db.xml files first and run Skyscraper afterwards, it will have broken the db.xml files on the destinationsince it will disregard all media it can't find and then resave the db.xml again without the media it couldn't find. So if I were you I'd start over completely and just copy the entire cache folder over before running Skyscraper on the other machine.

**Comment by waugh-hoo** [comment]
So the "box art" data is all contained in the cache folder then?  I copied it as stated, but it does not appear to be pulling the box art.  I'll keep at it.

**Comment by muldjord** [comment]
That's probably because you didn't copy all of it in one go. _Everything_ is inside the cache folder. Nothing is located elsewhere.

**Comment by waugh-hoo** [comment]
okay... I'm pulling the FULL cache file from the other build and will drop it in un touched.  I'll then re-run skyscraper on the new machine and just generate the games list then?  I'll report back.  Thanks for the hand holding (in a bug report).

**Comment by muldjord** [comment]
Yes, exactly. :)

**Comment by waugh-hoo** [comment]
Sounds good.  The only twist is that there are two platforms that won't be in the new build (or maybe it is just one).  Will clearing the cache data for those platforms make it work correctly?  (the roms folder for those two platforms do not have any games/roms loaded).

**Comment by muldjord** [comment]
You can remove the individual platform folders inside of the cache folder without any problems. Or you can use the command-line to do it for you as documented here: https://github.com/muldjord/skyscraper/blob/master/docs/CLIHELP.md#--cache-purgekeywordmodule-andor-type

**Comment by waugh-hoo** [comment]
understood.  Just moving the cache folder now to the new platform and will then rebuild the gamelist.

**Comment by waugh-hoo** [comment]
So I rebuilt the gamelist on the new box (for one platform only as I'm trying to verify my technique works).  The gamelist found all the box art etc, but when I reboot, none of the artwork shows up.  It shows just game list without the associated atwork.  I'm still trying things... if you have somewhere to look I'd be happy to explore.

**Comment by muldjord** [comment]
Did you remember to quit the frontend while generating the gamelist? Otherwise it is overwritten when it restarts. That's probably the most important thing to remember when working with any game list generation for any frontend. Make sure you aren't running the frontend at the same time as it _will_ cause issues.

**Comment by waugh-hoo** [comment]
It is a MintyPi (gaming rig in an altoids tin) so I'm doing it all over SSH.  It did ask me that question when I gave it the command to reboot.  Before running the setup file, I quit emulation station which took the rig to the terminal window.  I then connect via SSH to build the game list and made the command to reboot.  I assumed that quitting emulation station on the retropie screen was what I needed to do.  Is there something else I need to do to quit the front end?

**Comment by waugh-hoo** [comment]
also, FTPing into the new rig and downloading the xml file (gamelist), it does show the path and file name for the cover art.

**Comment by waugh-hoo** [comment]
Here's a copy of the XML doc...
[gamelist sample.txt](https://github.com/muldjord/skyscraper/files/4082031/gamelist.sample.txt)

**Comment by muldjord** [comment]
Looks fine. I think this is getting a little out of scope (this seems to be an ES issue really or maybe some media export paths you've configured that aren't the same on the two systems, I can't really know) so I'm gonna leave it to you from here. If everything is set up as documented you are basically looking at something that is not Skyscraper related or some custom config for paths you have made that I am not aware of.

**Comment by waugh-hoo** [comment]
Thanks for providing so much support despite it being a bit out of scope.  I'm with you... something is not quite the same between the two builds.  The MintyPi creaters have a specific image for their creation that is slightly different.  I suspect that the issue lies in that issue.  I was hoping that I could transfer the raw images and videos from the working build to a file on the new build and let sky scraper build it from there.  I'm just not quite sure how I would do that (it's like someone that speaks a tiny bit of a language and can't figure out what someone is saying but knows they are talking about food).  I'll keep messing about.  Thanks again for all the support.

**Comment by waugh-hoo** [comment]
well... I'll be dipped.  On a whim, I decided to see if the API was up and the skyscraper is currently downloading things from the site on the new box!  I am only doing one system at the moment as I want to make sure things work before I sit here for hours trying.  This now begs the question, will the current build files I have in the CACHE folders for those emulators get over written when I use skyscraper to build them pulling data from ScreenScraper.fr or should I delete those cache folders and let skyscraper build them anew?

**Comment by muldjord** [comment]
I don't quite understand the question but this might help: https://github.com/muldjord/skyscraper/blob/master/docs/CLIHELP.md#--cache-mergefolder

I'm gonna close this now and leave you to read the documentation. It is a bit much for me to provide support at this level - hope you understand. Good luck with it!

**Comment by waugh-hoo** [comment]
I totally understand and appreciate the level of support I got.  I will admit to continuing to ask until I got a no ;-)  I'll read up and keep banging away at it.  So far it seems to be working on some of the existing scrapes I had.  

Thanks so much for building such an awesome product and sharing it with us.  I appreciate it.

**Comment by waugh-hoo** [comment]
So... to close the loop on what the issue ended up being.  In my retropie ROM folder for pcengine, I had an gameslist.xml file that was from a rom pack I assembled sometime ago.  This, as luck would turn out, was my test case emulator I was using.  Somehow, despite the emulation station gameslist.xml being correct, it was reading the one in the ROM folder which had an accurate games list, but not the paths for the images and box art.  I finally remembered it being there and removed it and rescraped that one emulator and now it is working correctly.  Thanks for all the help along the way.  Just thought I'd drop my solution in here in case someone else was having a similar issue.
